src/main/java/moze_intel/projecte/gameObjs/entity/EntityHomingArrow.java

```diff
@@ -1,105 +1,198 @@
+import net.minecraft.util.MathHelper;
+import net.minecraft.util.MovingObjectPosition;
+import net.minecraft.util.Vec3;
 import net.minecraft.world.World;
 
+import javax.vecmath.AxisAngle4d;
+import javax.vecmath.Matrix4d;
+import javax.vecmath.Matrix4f;
+import javax.vecmath.Quat4f;
+import javax.vecmath.Vector3d;
+import java.util.Collections;
+import java.util.Comparator;
 import java.util.List;
 
 public class EntityHomingArrow extends EntityArrow
 {
-	EntityLiving target;
-	World world;
+	private static final int DW_TARGET_ID = 31;
+	private static final int NO_TARGET = -1;
 
-	private void init(World world)
-	{
-		this.world = world;
-	}
+	private int newTargetCooldown = 0;
 
 	public EntityHomingArrow(World world)
 	{
 		super(world);
-		init(world);
 	}
 
 	public EntityHomingArrow(World world, EntityLivingBase par2, float par3) 
 	{
 		super(world, par2, par3);
-		init(world);
 	}
-	
+
+	@Override
+	public void entityInit()
+	{
+		super.entityInit();
+		dataWatcher.addObject(DW_TARGET_ID, NO_TARGET); // Target entity id
+	}
+
 	@Override
 	public void onUpdate()
 	{
-		//TODO Create proper custom arrow. This one is duplicating because of the super call for onUpdate();
-		super.onUpdate();
+		onEntityUpdate();
+		boolean inGround = WorldHelper.isArrowInGround(this);
+		if (!worldObj.isRemote)
+		{
+			if (hasTarget() && (!getTarget().isEntityAlive() || inGround))
+			{
+				dataWatcher.updateObject(DW_TARGET_ID, NO_TARGET);
+				PELogger.logInfo("Removing target");
+			}
 
-		AxisAlignedBB box = this.boundingBox;
-		
-		if (target == null && !WorldHelper.isArrowInGround(this))
+			if (!hasTarget() && !inGround && newTargetCooldown <= 0)
+			{
+				PELogger.logInfo("Finding new target");
+				findNewTarget();
+			} else
+			{
+				newTargetCooldown--;
+			}
+		}
+
+		if (ticksExisted > 3 && hasTarget() && !WorldHelper.isArrowInGround(this))
 		{
-			AxisAlignedBB bBox = box.expand(8, 8, 8);
-			List<EntityLiving> list = this.worldObj.getEntitiesWithinAABB(EntityLiving.class, bBox);
+			AxisAlignedBB box = this.boundingBox;
+			worldObj.spawnParticle("flame", box.maxX, box.maxY, box.maxZ, 0.0D, 0.0D, 0.0D);
+			setIsCritical(true);
+			Entity target = getTarget();
 
-			double distance = 100000;
 
-			for (EntityLiving entity : list)
-			{
-				double toIt = distanceTo(entity);
+			Vector3d arrowLoc = new Vector3d(posX, posY, posZ);
+			Vector3d targetLoc = new Vector3d(target.posX, target.boundingBox.minY + target.height, target.posZ);
+			Vector3d lookVec = new Vector3d(targetLoc);
+			lookVec.sub(arrowLoc);
+
+			// double dotProduct = new Vector3d(arrowLoc).dot(targetLoc);
+			double theta = arrowLoc.angle(lookVec); //Math.acos(dotProduct / (arrowLoc.length() * targetLoc.length()));
+
+			Vector3d crossProduct = new Vector3d();
+			crossProduct.cross(arrowLoc, targetLoc);
+			crossProduct.normalize();
+
+			Matrix4d transform = new Matrix4d();
+			transform.set(new AxisAngle4d(crossProduct, theta * 0.65));
```

Comment:
I think instead of always rotating by 65% of the angle you want to apply a maximum turn rate for the arrow here.